File: src/decoders/holiday.ts
import type { Holiday } from "~/models";
import { decodePronoteDate } from "./pronote-date";

export const decodeHoliday = (holiday: any): Holiday => {
  return {
    id: holiday.N,
    name: holiday.L,
    startDate: decodePronoteDate(holiday.dateDebut.V),
    endDate: decodePronoteDate(holiday.dateFin.V)
  };
};

File: library/src/commonMain/kotlin/decoders/holiday.kt
package decoders

import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.jsonObject
import kotlinx.serialization.json.jsonPrimitive
import models.Holiday

fun decodeHoliday (holiday: JsonObject): Holiday {
    return Holiday(
        id = holiday["N"]!!.jsonPrimitive.content,
        name = holiday["L"]!!.jsonPrimitive.content,
        startDate = decodePronoteDate(holiday["dateDebut"]!!.jsonObject["V"]!!.jsonPrimitive.content),
        endDate = decodePronoteDate(holiday["dateFin"]!!.jsonObject["V"]!!.jsonPrimitive.content)
    )
}